packages/push/examples/complex_regression/main.rs

```diff
@@ -35,60 +36,91 @@ use crate::args::{Args, RunModel};
  * https://github.com/lspector/propeller/blob/71d378f49fdf88c14dda88387291c9c7be0f1277/src/propeller/problems/complex_regression.cljc
  */
 
-fn main() -> Result<()> {
+type Of64 = OrderedFloat<f64>;
+
+/// The target polynomial is (x^3 + 1)^3 + 1
+fn target_fn(input: Of64) -> Of64 {
+    let sub_expr = input * input * input + 1.0;
+    sub_expr * sub_expr * sub_expr + 1.0
+}
+
+fn build_push_state(
+    program: impl DoubleEndedIterator<Item = PushProgram> + ExactSizeIterator,
+    input: Of64,
+) -> PushState {
+    #[allow(clippy::unwrap_used)]
+    PushState::builder()
+        .with_max_stack_size(1000)
+        .with_program(program)
+        // This will return an error if the program is longer than the allowed
+        // max stack size.
+        // We arguably should check that and return an error here.
+        .unwrap()
+        .with_float_input("x", input)
+        .build()
+}
+
+fn score_program(
+    program: impl DoubleEndedIterator<Item = PushProgram> + ExactSizeIterator,
+    input: Of64,
+    expected_output: Of64,
+) -> Of64 {
     // The penalty value to use when an evolved program doesn't have an expected
     // "return" value on the appropriate stack at the end of its execution.
     const PENALTY_VALUE: f64 = 1_000.0;
 
+    let state = build_push_state(program, input);
+    #[allow(clippy::option_if_let_else)]
+    match state.run_to_completion() {
+        Ok(final_state) => OrderedFloat(
+            final_state
+                .stack::<Of64>()
+                .top()
+                .map_or(PENALTY_VALUE, |answer| (answer - expected_output).abs()),
+        ),
+        Err(_) => {
+            // Do some logging, perhaps?
+            OrderedFloat(PENALTY_VALUE)
+        }
+    }
+}
+
+fn score_genome(
+    genome: &Plushy,
+    training_cases: &Cases<Of64>,
+) -> TestResults<test_results::Error<Of64>> {
+    let program = Vec::<PushProgram>::from(genome.clone());
+    let errors: TestResults<test_results::Error<Of64>> = training_cases
+        .iter()
+        .map(|case| score_program(program.iter().cloned(), case.input, case.output))
+        .collect();
+    errors
+}
+
+fn main() -> Result<()> {
     let args = Args::parse();
 
     // Inputs from -4 (inclusive) to 4 (exclusive) in increments of 0.25.
-    let training_cases = (-4 * 4..4 * 4)
-        .map(|n| OrderedFloat(f64::from(n) / 4.0))
-        .collect::<Vec<_>>();
+    let training_inputs = (-4 * 4..4 * 4).map(|n| OrderedFloat(f64::from(n) / 4.0));
```

Comment:
```suggestion
    let training_inputs = (-4 * 4..4 * 4).map(|n| Of64::from(n) / 4.0);
```
I am not sure if this works but if it does it should be cleaner

Reply:
Worked great! Nice catch.